src/schema/read-schema.ts

```diff
@@ -7,13 +7,16 @@ import log from "../utils/logger";
 /**
  * Reads the current schema of the database
  */
-export default async function readSchema(): Promise<SchemaState> {
+export default async function readSchema(allowedDirectusTables: string): Promise<SchemaState> {
   log.message("loading", "Reading your database schema");
 
   const inspector = SchemaInspector(db());
+  const allowedTables = allowedDirectusTables.length ? allowedDirectusTables.split(',') : [];
 
   // Directus will take care of the system tables, so we exclude them
-  const columnDefs = (await inspector.columnInfo()).filter(col => !isDirectusTable(col.table));
+  const columnDefs = (await inspector.columnInfo()).filter(col =>
+    allowedDirectusTables.includes(col.table) || !isDirectusTable(col.table)
```

Comment:
I think it's best if we reverse that condition. We don't want to check all tables as it will never be true.
```suggestion
    !isDirectusTable(col.table) || allowedDirectusTables.includes(col.table)
```

Reply:
Yep, good point! thanks man